plugins/embed-optimizer/hooks.php

```diff
@@ -10,6 +10,36 @@
 	exit; // Exit if accessed directly.
 }
 
+/**
+ * Add hooks.
+ *
+ * @since n.e.x.t
+ */
+function embed_optimizer_add_hooks(): void {
+	add_action( 'wp_head', 'embed_optimizer_render_generator' );
+
+	if ( defined( 'OPTIMIZATION_DETECTIVE_VERSION' ) ) {
+		add_action( 'od_register_tag_visitors', 'embed_optimizer_register_tag_visitors', 10, 3 );
+	} else {
+		add_filter( 'embed_oembed_html', 'embed_optimizer_filter_oembed_html' );
+	}
+}
+add_action( 'init', 'embed_optimizer_add_hooks' );
+
+/**
+ * Registers the tag visitor for embeds.
+ *
+ * @since n.e.x.t
+ *
+ * @param OD_Tag_Visitor_Registry         $registry                     Tag visitor registry.
+ * @param OD_URL_Metrics_Group_Collection $url_metrics_group_collection URL Metrics Group Collection.
+ * @param OD_Link_Collection              $link_collection              Link Collection.
+ */
+function embed_optimizer_register_tag_visitors( OD_Tag_Visitor_Registry $registry, OD_URL_Metrics_Group_Collection $url_metrics_group_collection, OD_Link_Collection $link_collection ): void {
+	require_once __DIR__ . '/class-embed-optimizer-tag-visitor.php';
```

Comment:
This `require_once` could go into `load.php`. But there's no need to load it if it won't be used. I'm happy to move it however.

Reply:
maybe add an inline comment explaining why you are loading it here (to avoid loading it unnecessarily)?

Reply:
Added in e605df59